### PR TITLE
Add mapping for Gather operator; Fix pool_type in max_pooling

### PR DESCRIPTION
**Pull request by HeliWang**
## Description ##
Add mapping to Gather operator; Fix pool_type in max_pooling

### Changes ###
- Add mapping to Gather operator
- Fix pool_type in max_pooling

### CODE REVIEW

File: python/mxnet/contrib/onnx/_import/op_translations.py [hunk 1]
```diff
@@ -420,15 +424,15 @@ def avg_pooling(attrs, inputs, cls):
...
+    """ Max pooling"""
     new_attrs = translation_utils._fix_attribute_names(attrs,
                                                        {'kernel_shape': 'kernel',
                                                         'strides': 'stride',
                                                         'pads': 'pad',
                                                        })
 
     new_attrs = translation_utils._add_extra_attributes(new_attrs,
-                                                        {'pool_type': 'avg',
+                                                        {'pool_type': 'max',
```
Comment (rajanksin) [inline python/mxnet/contrib/onnx/_import/op_translations.py]: Thanks for catching,  but looking at the code looks like  'pool_type: 'max' is actually redundant code.  I think we can get rid of that

Comment (HeliWang) [reply python/mxnet/contrib/onnx/_import/op_translations.py]: True true the type is passed to _fix_pooling function. I will update to remove the pool_type field here.

**Comment by rajanksin** [comment]
Please also enable the test in :  incubator-mxnet/tests/python-pytest/onnx/onnx_backend_test.py
IMPLEMENTED_OPERATORS_TEST = [
'test_gather*'
]

Please make sure the onnx_backend_test.py passes.

**Comment by HeliWang** [comment]
I saw you have fixed the parameter of max_pooling so I made to revert to my "fix max_pooling" commit which makes it easier to merge with the updated base branch.

**Comment by HeliWang** [comment]
I have run the pytest onnx_backend_test.py and I got the following 7 failed cases:

OnnxBackendModelTest.test_MaxPool1d_cpu
>           padding = (0, padding[0], 0, padding[1])
E           IndexError: tuple index out of range

OnnxBackendModelTest.test_MaxPool1d_stride_cpu
>           padding = (0, padding[0], 0, padding[1])
E           IndexError: tuple index out of range

OnnxBackendModelTest.test_MaxPool2d_cpu
E           AssertionError: 
E           Not equal to tolerance rtol=0.001, atol=0

OnnxBackendModelTest.test_MaxPool3d_cpu
mxnet.base.MXNetError: Error in operator pad4: [00:11:40] src/operator/./pad-inl.h:211: Input shape vs padding spec mismatch.

OnnxBackendModelTest.test_MaxPool3d_stride_cpu
mxnet.base.MXNetError: Error in operator pad5: [00:11:40] src/operator/./pad-inl.h:211: Input shape vs padding spec mismatch.

OnnxBackendModelTest.test_MaxPool3d_stride_padding_cpu
mxnet.base.MXNetError: Error in operator pad6: [00:11:40] src/operator/./pad-inl.h:211: Input shape vs padding spec mismatch.

OnnxBackendModelTest.test_densenet121_cpu
FileNotFoundError: [Errno 2] No such file or directory: '/home/h379wang/.onnx/models/densenet121/model.pb'

Are the cases known for you? I have also add "test_gather" to IMPLEMENTED_OPERATORS_TEST and it seems that the ONNX backend test for gather operator has passed.

**Comment by rajanksin** [comment]
I am guessing you are using ONNX version 1.0.1.  The tests above  will pass with the latest master. We are in process of getting ONNX version bumped up too. Since your 'Gather' test passed, we should be good here.

**Pull request merged**